File: 1777-determine-if-two-strings-are-close/determine-if-two-strings-are-close.java
class Solution {
    public boolean closeStrings(String word1, String word2) {
        if(word1.length()!=word2.length()) return false;
        else{
        int a[]=new int[27];
        int b[]=new int[27];
        for(int i=0;i<word1.length();i++){
                char ch=word1.charAt(i);
                a[ch-97]++;            
        }
        for(int i=0;i<word1.length();i++){
                char ch=word2.charAt(i);
                if(a[ch-97]!=0) b[ch-97]++;
                else return false;            
        }
        Arrays.sort(a);
        Arrays.sort(b);
        for(int i=0;i<27;i++)
        {
            if(a[i]!=b[i])return false;
        }
        return true;
        }      
}
}

File: 1777-determine-if-two-strings-are-close/determine-if-two-strings-are-close.cpp
class Solution {
public:
    bool closeStrings(string word1, string word2) {
        if(word1.size()!=word2.size()) return false;
        vector<int> mp1(26, 0), mp2(26, 0);
        for(char c:word1) mp1[c-'a']++;
        for(char c:word2){
            if(!mp1[c-'a']) return false;
            else mp2[c-'a']++;
        }
        sort(mp1.begin(), mp1.end());
        sort(mp2.begin(), mp2.end());
        return mp1==mp2;
    }
};